tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -558,13 +558,34 @@ static void ExtendDestinationModel(std::span<const RFieldDescriptor *> newFields
         (newFields.size() > 1 ? "them" : "it"), mergeData.fNumDstEntries);
 
    changeset.fAddedFields.reserve(newFields.size());
+   // First add all non-projected fields...
    for (const auto *fieldDesc : newFields) {
-      auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
-      if (fieldDesc->IsProjectedField())
+      if (!fieldDesc->IsProjectedField()) {
+         auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
+         changeset.AddField(std::move(field));
+      }
+   }
+   // ...then add all projected fields.
+   Internal::RProjectedFields::FieldMap_t fieldMap;
+   for (const auto *fieldDesc : newFields) {
+      if (fieldDesc->IsProjectedField()) {
+         auto field = fieldDesc->CreateField(*mergeData.fSrcDescriptor);
+         const auto sourceId = fieldDesc->GetProjectionSourceId();
+         const auto &sourceField = dstModel.GetConstField(mergeData.fSrcDescriptor->GetQualifiedFieldName(sourceId));
+         fieldMap[field.get()] = &sourceField;
+         const auto &subFieldIds = fieldDesc->GetLinkIds();
+         int i = 0;
+         for (const auto &subfield : *field) {
+            const auto subFieldId = subFieldIds[i++];
```

Comment:
I don't think that's correct. The `*field` iterator walks through the entire subfield hierarchy, but the `GetLinkIds()` only gets the direct children. I think we should test this with an alias field of a `vector<CustomStruct>` field.